 Do not scan a file which cannot be read

When the media scanner is invoked to scan a file which it cannot access
(e.g. one in the app-private directories) the related database entry
will be created anyway, which doesn't make sense. This results in having
broken entries showed up in e.g. Album app.

This fix is to prevent scanning files which are inaccessible.

diff --git a/media/java/android/media/MediaScanner.java b/media/java/android/media/MediaScanner.java
index 1008a50..080f6b1 100644
--- a/media/java/android/media/MediaScanner.java
+++ b/media/java/android/media/MediaScanner.java
@@ -1457,7 +1457,7 @@ public Uri scanSingleFile(String path, String volumeName, String mimeType) {
             prescan(path, true);
 
             File file = new File(path);
-            if (!file.exists()) {
+            if (!file.exists() || !file.canRead()) {
                 return null;
             }
 
